### PR TITLE
Laboratory work #2, Ulyana Nagornaya - 22FPL2

### PR DESCRIPTION
**Pull request by Ulyana-Nagornaya**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
👀

File: lab_2_tokenize_by_bpe/main.py [hunk 1]
```diff
@@ -3,7 +3,6 @@
 BPE and machine translation evaluation
 """
 
-
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:6]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 2]
```diff
@@ -26,7 +36,23 @@ def collect_frequencies(
...
+    words = text.split()
+    tokenized_words = []
+    dict_of_freq = {}
+
+    for word in words:
+        freq = words.count(word)
+        word = prepare_word(word, start_of_word, end_of_word)
+        if word is None:
+            return None
+        tokenized_words.append(word)
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:53]: for what?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_tokenize_by_bpe/main.py [hunk 3]
```diff
@@ -47,7 +87,24 @@ def merge_tokens(
...
     """
+    if not isinstance(word_frequencies, dict) \
+        or not isinstance(pair, tuple):
+        return None
+
+    processed_dict = {}
+    couple_of_items = pair[0] + pair[1]
 
+    for word, freq in word_frequencies.items():
+        if f"'{pair[0]}', '{pair[1]}'" in str(word):
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:98]: woow... Please, simplify

File: lab_2_tokenize_by_bpe/target_score.txt [hunk 4]
```diff
@@ -1 +1 @@
-0
+10
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/target_score.txt:1]: very good

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_tokenize_by_bpe/main.py (hunk 3 above)
Comment (Ulyana-Nagornaya) [reply lab_2_tokenize_by_bpe/main.py]: I`ve changed

File: lab_2_tokenize_by_bpe/main.py [hunk 5]
```diff
@@ -69,7 +147,27 @@ def get_vocabulary(
...
+        return None
+
+    dict_of_ident = {}
+    diff_tokens = set()
+    diff_tokens.add(unknown_token)
+
+    for key in word_frequencies:
+        for word in key:
+            diff_tokens.add(word)
+            for token in word:
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:161]: `.update`

File: lab_2_tokenize_by_bpe/main.py [hunk 6]
```diff
@@ -81,7 +179,26 @@ def decode(
...
+
+    decoded_sequence = ''
+    for ident in encoded_text:
+        list_for_token = [key for key, value in vocabulary.items() if value == ident]
 
+        for token in list_for_token:
+            if end_of_word_token:
+                if end_of_word_token in token:
+                    end_index = len(end_of_word_token)
+                    decoded_sequence += end_of_word_token[end_index:] + ' '
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:196]: is it replace?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Very nice

**Pull request closed without merging**